lua/neotest/consumers/watch/init.lua

```diff
@@ -32,14 +32,21 @@ neotest.watch = {}
 local function get_valid_client(bufnr)
   local clients = nio.lsp.get_clients({ bufnr = bufnr })
   for _, client in ipairs(clients) do
-    local has_definition_support = client.supports_method
-        and client.supports_method("textDocument/definition")
-      -- for compatibility with Neovim versions earlier than v0.10.1
-      or client.server_capabilities.definitionProvider
+    local has_definition_support
+    if client.server_capabilities then
+      -- for compatibility with Neovim versions earlier and equal to v0.10.1
+      has_definition_support = client.server_capabilities.definitionProvider
+    elseif type(client.supports_method) == "function" then
```

Comment:
1. Actually this should be using client.supports_method.textDocument_definition as a function. Like [here](https://github.com/nvim-neotest/nvim-nio/blob/a428f309119086dc78dd4b19306d2d67be884eee/tests/lsp_spec.lua#L97-L113).

2. Actually type(client.supports_method.textDocument_definition) will always be a `function`, so we should check nvim version instead.

3. Actually supports_method was already included in nvim v0.10.1 . See https://github.com/neovim/neovim/commit/ddd92a70d2aab5247895e89abaaa79c62ba7dbb4 .
    
Something like this:
```lua
    local has_definition_support
    if nio.fn.has("nvim-0.10.1") == 0 then
      -- for compatibility with Neovim versions earlier than v0.10.1
      has_definition_support = client.server_capabilities ~= nil and client.server_capabilities.definitionProvider
    else
      has_definition_support = client.supports_method.textDocument_definition({ bufnr = bufnr })
    end



```

Reply:
Yeah, you're absolutely right. Just tested it on 0.10.1, just in case, fixes the problem. Looks better as well, I think.